### PR TITLE
Add delete by tag function

### PR DESCRIPTION
**Pull request by bangyiwu**
Allows the users to delete all users with a particular tag

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/contacts/DeleteContactByTagCommand.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     * Model deletes persons with the user-input tag recursively to prevent concurrent modification exception.
+     */
+    public void deleteByRecursion(Model model) {
+        Set<Person> personsToDelete = model.getPersonsWithTag(tagForDeletion);
+        Iterator<Person> personIterator = personsToDelete.iterator();
+        if (personIterator.hasNext()) {
+            model.deletePerson(personIterator.next());
+            deleteByRecursion(model);
+        }
+    }
```
Comment (chan-j-d) [inline src/main/java/seedu/address/logic/commands/contacts/DeleteContactByTagCommand.java]: `ContactTagIntegrationManager` has a method for deleting a tag and all contacts with the tag. I think it's called `deleteTagAndDirectContacts` method that does that. I think you should use that instead as it properly preserves the tag relations in the `tagTree` as well. Can implement it in `Model`

Comment (bangyiwu) [reply src/main/java/seedu/address/logic/commands/contacts/DeleteContactByTagCommand.java]: okay will do

File: src/main/java/seedu/address/logic/commands/CommandWord.java [hunk 2]
```diff
@@ -21,6 +22,9 @@
     LIST("list"),
     VIEW("view"),
     SORT("sort"),
+    REMIND("remind"),
+    REMINDER("listr"),
+    SHOW_REMINDER("showr"),
```
Comment (LinkedInk) [inline src/main/java/seedu/address/logic/commands/CommandWord.java:27]: Merged from another branch? They should'nt ber here for this PR. This means that the other PR has to be approved first before this. Else might as well have this PR only.

**Comment by LinkedInk** [comment]
Do not review, this PR currently still has the old and outdated files.

**Pull request closed without merging**